llvm/tools/llvm-profgen/CSPreInliner.cpp

```diff
@@ -179,11 +179,8 @@ bool CSPreInliner::shouldInline(ProfiledInlineCandidate &Candidate) {
         (NormalizationUpperBound - NormalizationLowerBound);
     if (NormalizedHotness > 1.0)
       NormalizedHotness = 1.0;
-    // Add 1 to ensure hot callsites get a non-zero threshold, which could
-    // happen when SampleColdCallSiteThreshold is 0. This is when we do not
-    // want any inlining for cold callsites.
     SampleThreshold = SampleHotCallSiteThreshold * NormalizedHotness * 100 +
-                      SampleColdCallSiteThreshold + 1;
```

Comment:
Looking at it more closely, this actually makes `SampleThreshold` smaller by 100 (`CSPreinlMultiplierForPrevInl`) for the hot path.. which is not intended for the original change? 

Maybe we should just +1 for the cold path..

Reply:
I encountered similar issue in the past and I went for an alternative that is to increase function size by 1 in `CSPreInliner::processFunction` . Perf win wasn't obvious though.